fsw/cfe-core/src/sb/cfe_sb_buf.c

```diff
@@ -48,82 +97,60 @@
 **    by the SB to dynamically allocate memory to hold the message and a buffer
 **    descriptor associated with the message during the sending of a message.
 **
+**  Note:
+**    This must only be invoked while holding the SB global lock
+**
 **  Arguments:
-**    msgId        : Message ID
-**    size         : Size of the buffer in bytes.
+**    MaxMsgSize         : Size of the buffer content area in bytes.
 **
 **  Return:
 **    Pointer to the buffer descriptor for the new buffer, or NULL if the buffer
 **    could not be allocated.
 */
 
-CFE_SB_BufferD_t * CFE_SB_GetBufferFromPool(CFE_SB_MsgId_t MsgId, size_t Size) {
-   int32                stat1;
-   uint8               *address = NULL;
-   CFE_SB_BufferD_t    *bd = NULL;
+CFE_SB_BufferD_t *CFE_SB_GetBufferFromPool(size_t MaxMsgSize)
+{
+    int32             stat1;
+    size_t            AllocSize;
+    CFE_SB_BufferD_t *bd = NULL;
+
+    /* The allocation needs to include enough space for the descriptor object */
+    AllocSize = MaxMsgSize + CFE_SB_BUFFERD_CONTENT_OFFSET;
 
     /* Allocate a new buffer descriptor from the SB memory pool.*/
-    stat1 = CFE_ES_GetPoolBuf((CFE_ES_MemPoolBuf_t*)&bd, CFE_SB.Mem.PoolHdl,  Size + sizeof(CFE_SB_BufferD_t));
-    if(stat1 < 0){
+    stat1 = CFE_ES_GetPoolBuf((CFE_ES_MemPoolBuf_t *)&bd, CFE_SB.Mem.PoolHdl, AllocSize);
+    if (stat1 < 0)
+    {
         return NULL;
     }
 
     /* increment the number of buffers in use and adjust the high water mark if needed */
     CFE_SB.StatTlmMsg.Payload.SBBuffersInUse++;
-    if(CFE_SB.StatTlmMsg.Payload.SBBuffersInUse > CFE_SB.StatTlmMsg.Payload.PeakSBBuffersInUse){
+    if (CFE_SB.StatTlmMsg.Payload.SBBuffersInUse > CFE_SB.StatTlmMsg.Payload.PeakSBBuffersInUse)
+    {
         CFE_SB.StatTlmMsg.Payload.PeakSBBuffersInUse = CFE_SB.StatTlmMsg.Payload.SBBuffersInUse;
-    }/* end if */
+    } /* end if */
 
     /* Add the size of the actual buffer to the memory-in-use ctr and */
     /* adjust the high water mark if needed */
-    CFE_SB.StatTlmMsg.Payload.MemInUse+=stat1;
-    if(CFE_SB.StatTlmMsg.Payload.MemInUse > CFE_SB.StatTlmMsg.Payload.PeakMemInUse){
+    CFE_SB.StatTlmMsg.Payload.MemInUse += AllocSize;
+    if (CFE_SB.StatTlmMsg.Payload.MemInUse > CFE_SB.StatTlmMsg.Payload.PeakMemInUse)
+    {
         CFE_SB.StatTlmMsg.Payload.PeakMemInUse = CFE_SB.StatTlmMsg.Payload.MemInUse;
-    }/* end if */
-
-    /* first set ptr to actual msg buffer the same as ptr to descriptor */
-    address = (uint8 *)bd;
-
-    /* increment actual msg buffer ptr beyond the descriptor */
-    address += sizeof(CFE_SB_BufferD_t);
+    } /* end if */
 
     /* Initialize the buffer descriptor structure. */
-    bd->MsgId     = MsgId;
-    bd->UseCount  = 1;
-    bd->Size      = Size;
-    bd->Buffer    = (CFE_SB_Buffer_t *)address;
```

Comment:
Good to see this updated. This line in the current `main` branch increases alignment requirements on ARM32 when going from the `uint8*` to `CFE_SB_Buffer_t*`.